src/main/java/com/indeed/jiraactions/api/ApiCaller.java

```diff
@@ -3,173 +3,94 @@
 import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.indeed.jiraactions.JiraActionsIndexBuilderConfig;
+
+import okhttp3.ConnectionPool;
+import okhttp3.Headers;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import okhttp3.ResponseBody;
 import org.apache.commons.codec.binary.Base64;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import javax.net.ssl.HttpsURLConnection;
-import java.io.BufferedReader;
 import java.io.IOException;
-import java.io.InputStream;
-import java.io.InputStreamReader;
-import java.net.URL;
-import java.util.List;
-import java.util.Map;
-import java.util.Objects;
+import java.util.concurrent.TimeUnit;
 
 public class ApiCaller {
     protected final JiraActionsIndexBuilderConfig config;
 
     private static final Logger log = LoggerFactory.getLogger(ApiCaller.class);
     private static final ObjectMapper objectMapper = new ObjectMapper();
+    private OkHttpClient client;
     private final String authentication;
-    private String jsessionId = null;
-    private String upstream = null;
-    private String cookies = "";
-    private String pinnedNode = null;
 
     public ApiCaller(final JiraActionsIndexBuilderConfig config) {
         this.config = config;
         this.authentication = getBasicAuth();
+        this.client = new OkHttpClient.Builder()
+                .connectionPool(new ConnectionPool())
+                .connectTimeout(2, TimeUnit.MINUTES)
```

Comment:
Where did these values come from?

Also, putting one in MINUTES and one in MILLISECONDS is weird, especially when the one in MILLISECONDS could be specified in seconds.